game/src/Utility/StyleBoxWithSound.gd

```diff
@@ -0,0 +1,33 @@
+@tool
+extends StyleBox
+class_name StyleBoxWithSound
+
+@export
+var style_box : StyleBox:
+	get: return style_box
+	set(v):
+		style_box = v
+		emit_changed()
+
+@export
+var sound : AudioStream:
+	get: return sound
+	set(v):
+		sound = v
+		emit_changed()
+
+func _get_draw_rect(rect : Rect2) -> Rect2:
+	if style_box == null: return Rect2()
+	return style_box._get_draw_rect(rect)
+
+func _draw(to_canvas_item : RID, rect : Rect2) -> void:
```

Comment:
Interesting solution certainly, not entirely sure about meshing audio with draw functionality but it works for now.

Reply:
Yeah, its kinda the only way to get audio functionality into the styles, which is easier to manage then trying to inject into every pressed signal to output a sound somehow.